### ISSUE TITLE
Split `irods_common_core` and `irods_lib_core_[client|server]` into separate directories.

### ISSUE THREAD

**Issue by SwooshyCueb** [irods/irods#7738]
The `irods_common_core` and `irods_lib_core_[client|server]` targets are defined in `lib/core`:
https://github.com/irods/irods/blob/e4b4e8a7266360bb38ee97d7f2eda69249567dff/lib/core/CMakeLists.txt#L19-L20
https://github.com/irods/irods/blob/e4b4e8a7266360bb38ee97d7f2eda69249567dff/lib/core/CMakeLists.txt#L202-L203

Let's split them into separate directories.

Some decisions will need to be made:
- Whether `irods_common_core` or `irods_lib_core_[client|server]` will remain in the core directory
- What the new directory shall be named
- What the new target names shall be

There may also be headers that do not clearly belong to one or the other.

**Comment by trel** [comment]
presumably we have a common because there are things for both?   why split them?   because composability is easy/good?

**Comment by SwooshyCueb** [comment]
`irods_common_core` is an object library used to make `libirods_common`.
`irods_lib_core_client` and `irods_lib_core_server` are two object libraries made from the same source files used to make `libirods_client` and `libirods_server`, respectively.
It doesn't make sense to me to have them live in the same directory.

**Comment by trel** [comment]
fair enough - if we're building the same targets in the end... nobody downstream needs to know/care i presume...

**Comment by SwooshyCueb** [comment]
Duplicate of #6207
